test/array.cpp

```diff
@@ -418,4 +418,15 @@ TEST(array_negative_strides) {
       [&](const dense_array<int, 2>::index_type& i) { ASSERT_EQ(b(i), std::get<1>(i)); });
 }
 
+TEST(array_move_reinterpret_shape) {
+  array_of_rank<int, 3> a({9, {0, 5, 10}, 1});
```

Comment:
Can you use designated initializers or at least comments on what the fields mean? It's impossible for me to remember the order in {0, 5, 10}.

Reply:
This is a fair point but this pattern is used probably *hundreds* of times in the tests. If we're going to fix it, we should fix it everywhere (in a separate PR), and I'm not sure we should.

BTW it's min, extent, stride. In this case, the test is constructing an array with a padding of one element after each row. The test verifies that move_reinterpret_shape preserves this padding (if it didn't, the pattern check would be incorrect).